File: content/problems/2826-sorting-three-groups/solutions/three-ending-group-dp/Solution.java
class Solution {
    public int minimumOperations(List<Integer> nums){
        int one=0,two=0,three=0;
        for(int value:nums){
            int nextOne=one+(value==1?0:1);
            int nextTwo=Math.min(one,two)+(value==2?0:1);
            int nextThree=Math.min(one,Math.min(two,three))+(value==3?0:1);
            one=nextOne;
            two=nextTwo;
            three=nextThree;
        }
        return Math.min(one,Math.min(two,three));
    }
}

File: content/problems/2826-sorting-three-groups/solutions/three-ending-group-dp/Solution.kt
class Solution {
    fun minimumOperations(nums:List<Int>):Int{
        var one=0;
        var two=0;
        var three=0;
        for(value in nums){
            val nextOne=one+if(value==1)0 else 1;
            val nextTwo=minOf(one,two)+if(value==2)0 else 1;
            val nextThree=minOf(one,two,three)+if(value==3)0 else 1;
            one=nextOne;
            two=nextTwo;
            three=nextThree
        };
        return minOf(one,two,three)
    }
}
